include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,127 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#define RANGES_V3_VIEW_LINEAR_DISTRIBUTE_HPP
+#include <type_traits>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/static_const.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-views
+        /// @{
+
+        /// Distributes `n >= 2` values linearly in the closed interval [`from`, `to`].
+        template<typename T>
+        struct linear_distribute_view
+          : view_facade<linear_distribute_view<T>, finite>
+        {
+            static_assert(std::is_arithmetic<T>(),
+                          "T must model the Arithmetic<T> concept; that is, "
+                          "it must be one of the native integer or floating-point types");
+        private:
+            friend range_access;
+
+            T from_, to_, spacing_;
+
+            constexpr T read() const
+            {
+                return from_;
+            }
+            RANGES_CXX14_CONSTEXPR void next()
+            {
+                from_ += spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_integral<T>())
+            constexpr bool equal(default_sentinel) const
+            {
+                return from_ == to_ + spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_floating_point<T>())
+            constexpr bool equal(default_sentinel) const
+            {
+                return from_ > to_;
+            }
+            RANGES_CXX14_CONSTEXPR void prev()
+            {
+                from_ -= spacing_;
+            }
+            RANGES_CXX14_CONSTEXPR void advance(std::ptrdiff_t n)
+            {
+                RANGES_EXPECT(std::numeric_limits<T>::max() / spacing_ >= n);
+                RANGES_EXPECT(to_ - from_ >= n * spacing_);
+                from_ += n * spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_integral<T>())
+            constexpr std::ptrdiff_t distance_to(linear_distribute_view const &that) const
+            {
+                return (that.from_ - from_) / spacing_;
+            }
+            CONCEPT_REQUIRES(std::is_floating_point<T>())
+            constexpr std::ptrdiff_t distance_to(linear_distribute_view const &that) const
+            {
+                return static_cast<std::ptrdiff_t>(std::floor((that.from_ - from_) / spacing_));
```

Comment:
I think this is a `ForwardRange`. Otherwise, the corrections for errors at each step gives you different answers. I suggest removing `prev`, `advance`, and `distance_to`, at least for floating point types.